code/controllers/subsystem/SSair.dm

```diff
@@ -245,109 +197,52 @@ SUBSYSTEM_DEF(air)
 		if(MC_TICK_CHECK)
 			return
 
-/datum/controller/subsystem/air/proc/process_high_pressure_delta(resumed = 0)
-	while(length(high_pressure_delta))
-		var/turf/simulated/T = high_pressure_delta[length(high_pressure_delta)]
-		high_pressure_delta.len--
-		T.high_pressure_movements()
-		T.pressure_difference = 0
-		if(MC_TICK_CHECK)
-			return
-
-/datum/controller/subsystem/air/proc/process_active_turfs(resumed = 0)
-	//cache for sanic speed
-	var/fire_count = times_fired
+/datum/controller/subsystem/air/proc/process_interesting_tiles(resumed = 0)
 	if(!resumed)
-		src.currentrun = active_turfs.Copy()
+		// Fetch the list of interesting tiles from Rust.
+		src.currentrun = get_interesting_atmos_tiles()
+		interesting_tile_count = length(src.currentrun)
 	//cache for sanic speed (lists are references anyways)
 	var/list/currentrun = src.currentrun
 	while(length(currentrun))
-		var/turf/simulated/T = currentrun[length(currentrun)]
+		// Pop a tile off the list.
+		var/list/interesting_tile = currentrun[length(currentrun)]
 		currentrun.len--
-		if(T)
-			T.process_cell(fire_count)
-		if(MC_TICK_CHECK)
-			return
 
-/datum/controller/subsystem/air/proc/process_excited_groups(resumed = 0)
-	if(!resumed)
-		src.currentrun = excited_groups.Copy()
-	//cache for sanic speed (lists are references anyways)
-	var/list/currentrun = src.currentrun
-	while(length(currentrun))
-		var/datum/excited_group/EG = currentrun[length(currentrun)]
-		currentrun.len--
-		EG.breakdown_cooldown++
-		if(EG.breakdown_cooldown == 10)
-			EG.self_breakdown()
-		else if(EG.breakdown_cooldown >= 20)
-			EG.dismantle()
+		var/x = interesting_tile[1]
+		var/y = interesting_tile[2]
+		var/z = interesting_tile[3]
+		var/x_flow = interesting_tile[4]
+		var/y_flow = interesting_tile[5]
+		var/fuel_burnt = interesting_tile[6]
+
+		var/turf/simulated/T = locate(x, y, z)
+		if(!istype(T))
+			if(MC_TICK_CHECK)
+				return
+			continue
```

Comment:
```suggestion
		var/turf/simulated/T = locate(x, y, z)
		if(!istype(T))
			if(MC_TICK_CHECK)
				return
			continue
```
I don't feel great about this. Locate can be quirky and that MC check scares me.

Reply:
As far as I can tell, locate(x, y, z) is *the* way to get a tile by coordinates, so I'm not sure what else to do.

What about the MC check bugs you?